datumaro/components/visualizer.py

```diff
@@ -192,14 +193,17 @@ def _sort_key(ann: Annotation):
     def vis_gallery(
         self,
         ids: List[Union[str, DatasetItem]],
-        subset: Optional[str] = None,
+        subset: Optional[Union[str, List[str]]] = None,
         grid_size: Tuple[Optional[int], Optional[int]] = (None, None),
     ) -> Figure:
         nrows, ncols = _infer_grid_size(len(ids), grid_size)
         fig, axs = plt.subplots(nrows, ncols, figsize=self.figsize)
 
-        for dataset_id, ax in zip(ids, axs.flatten()):
-            self.vis_one_sample(dataset_id, subset, ax)
```

Comment:
Would you add `assert isinstance(subset, list) and len(ids) == len(subset)` at this line?